src/avro/src/schema.rs

```diff
@@ -1155,6 +1155,13 @@ impl SchemaParser {
             .get("size")
             .and_then(|v| v.as_i64())
             .ok_or_else(|| ParseSchemaError::new("No `size` in fixed"))?;
+        if size <= 0 {
```

Comment:
after speaking with @umanwizard: the Avro spec does not require this to be non-zero/non-negative, but there's no message that could be encoded with those sizes

this fixes a `attempt to multiply with overflow` panic from the following test schema:
```
        // Decimal logical type
        r#"{"type": "fixed", "logicalType": "decimal", "name": "TestDecimal2", "precision": 2, "scale": 2, "size": -2}"#,
```